static/app/components/events/interfaces/spans/spanTree.tsx

```diff
@@ -20,24 +31,118 @@ import {
   FilterSpans,
   GroupType,
   ParsedTraceType,
+  SpanTreeNode,
+  SpanTreeNodeType,
   SpanType,
 } from './types';
-import {getSpanID, getSpanOperation, setSpansOnTransaction} from './utils';
+import {
+  getSpanID,
+  getSpanOperation,
+  isGapSpan,
+  setSpansOnTransaction,
+  spanTargetHash,
+} from './utils';
 import WaterfallModel from './waterfallModel';
 
 type PropType = ScrollbarManagerChildrenProps & {
   dragProps: DragManagerChildrenProps;
   filterSpans: FilterSpans | undefined;
   organization: Organization;
+  spanContextProps: SpanContext.SpanContextProps;
   spans: EnhancedProcessedSpanType[];
   traceViewRef: React.RefObject<HTMLDivElement>;
   waterfallModel: WaterfallModel;
   focusedSpanIds?: Set<string>;
 };
 
+const cache = new CellMeasurerCache({
+  fixedWidth: true,
+
+  defaultHeight: ROW_HEIGHT,
+  minHeight: ROW_HEIGHT,
+});
+
+const listRef = React.createRef<ReactVirtualizedList>();
+
 class SpanTree extends Component<PropType> {
   componentDidMount() {
     setSpansOnTransaction(this.props.spans.length);
+
+    if (location.hash) {
+      const {spans} = this.props;
+
+      // This reducer searches for the index of the anchored span.
+      // It's possible that one of the spans within an autogroup is anchored, so we need to search
+      // for the index within the autogroupings as well.
+      const {finalIndex, isIndexFound} = spans.reduce(
```

Comment:
tbh this reducer might be overcomplicated for what it is doing... 

Let me know if you don't like it and I'll see if I can make it better lol